cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -201,10 +202,28 @@ private void stopWatchWait(JsonArray externalAuthenticationURIs) throws Exceptio
       port = configuration.getInt(Constants.Security.AUTH_SERVER_BIND_PORT);
     }
 
+    String announceURLs = configuration.get(Constants.Security.AUTH_SERVER_ANNOUNCE_URLS);
```

Comment:
Use `configuration.getTrimmedStrings` instead. Also, an empty array will be returned instead of null.